src/shapes/polyline.class.ts

```diff
@@ -79,56 +108,104 @@ import { makeBoundingBoxFromPoints } from '../util/misc/boundingBoxFromPoints';
-              y: calcDim.top - this.strokeWidth / 2 + correctSize / 2,
-            },
-            'left',
-            'top',
-            this.originX,
-            this.originY
-          );
-        }
-        if (typeof options.left === 'undefined') {
-          this.left = options.fromSVG ? calcDim.left : correctLeftTop.x;
-        }
-        if (typeof options.top === 'undefined') {
-          this.top = options.fromSVG ? calcDim.top : correctLeftTop.y;
+      /**
+       * Calculate the polygon bounding box
+       * @private
+       */
+      _calcDimensions: function () {
+        const points = this.exactBoundingBox
+          ? this._projectStrokeOnPoints().map(
+              (projection) => projection.projectedPoint
+            )
+          : this.points;
+        if (points.length === 0) {
+          return {
+            left: 0,
+            top: 0,
+            width: 0,
+            height: 0,
+            pathOffset: new Point(),
+          };
         }
-        this.pathOffset = {
-          x: calcDim.left + this.width / 2 + correctSize / 2,
-          y: calcDim.top + this.height / 2 + correctSize / 2,
+        const bbox = makeBoundingBoxFromPoints(points);
+        const bboxNoStroke = makeBoundingBoxFromPoints(this.points);
+        const offsetX = bbox.left + bbox.width / 2,
+          offsetY = bbox.top + bbox.height / 2;
+        const pathOffsetX =
+          offsetX - offsetY * Math.tan(degreesToRadians(this.skewX));
+        const pathOffsetY =
+          offsetY - pathOffsetX * Math.tan(degreesToRadians(this.skewY));
+        // TODO: remove next line
+        const legacyCorrection =
+          !this.fromSVG && !this.exactBoundingBox ? this.strokeWidth / 2 : 0;
+        return {
+          ...bbox,
+          left: bbox.left - legacyCorrection,
+          top: bbox.top - legacyCorrection,
+          pathOffset: new Point(pathOffsetX, pathOffsetY),
+          strokeOffset: new Point(bboxNoStroke.left, bboxNoStroke.top).subtract(
+            bbox.left,
+            bbox.top
+          ),
         };
       },
 
       /**
-       * Calculate the polygon min and max point from points array,
-       * returning an object with left, top, width, height to measure the
-       * polygon size
-       * @return {Object} object.left X coordinate of the polygon leftmost point
-       * @return {Object} object.top Y coordinate of the polygon topmost point
-       * @return {Object} object.width distance between X coordinates of the polygon leftmost and rightmost point
-       * @return {Object} object.height distance between Y coordinates of the polygon topmost and bottommost point
+       * @returns {Point} top left position of the bounding box, useful for complementary positioning
+       */
+      setDimensions: function () {
+        const { left, top, width, height, pathOffset, strokeOffset } =
+          this._calcDimensions();
+        this.set({ width, height, pathOffset, strokeOffset });
```

Comment:
i kind of get that `strokeOffset` is the value that tells us in which direction the bounding box move most of its new space, becuase that was always one of the things that this kind of new calculation was adding, the center of the shape with and without stroke is no more the same and this strokeOffset kind of represent that.
But where are we using it? I can't see use it in the PR nor in the codebase.

Reply:
Trash my comments entirely in favor of 
f158267

Reply:
github mixes comments sometimes ( it does it to me ).
Indeed this answer is for the other thread.
strokeOffset still a mistery? or your answer was related and i don't see how?

Reply:
Related to the other thread indeed.
Can't find it anymore. Too much chat.

Don't understand your questions

Reply:
i ll try easier, at the cost of looking unpolite.

Show me where strokeOffset is used.
i see it calculated, not used.

Reply:
As I wrote in the thread it is meant for the dev.
Think of using originX/Y in a case of a large miter for example. The poly might shift a lot because of this miter. So a dev might want to adjust the position so the origin is in respect to the points instead of the entire stoked miter

Reply:
ok so i guessed right what is it, and is unused.
I'll think of it.